tests/integration/targets/test_azure_virtual_machine_with_public_ip/tasks/main.yml

```diff
@@ -45,54 +49,176 @@
       ansible.builtin.include_role:
         name: cloud.azure_ops.azure_virtual_machine_with_public_ip
       vars:
         azure_virtual_machine_with_public_ip_operation: delete
+        azure_virtual_machine_with_public_ip_delete_resource_group: true
+
+    # Verify Resource group deleted
+    - name: Get Resource group
+      azure.azcollection.azure_rm_resourcegroup_info:
+        name: "{{ azure_virtual_machine_with_public_ip_resource_group }}"
+      register: rg_info
+
+    - name: Ensure Resource group deleted
+      ansible.builtin.assert:
+        that:
+          - rg_info.resourcegroups | length == 0
+
+  always:
+    - name: Delete a Resource group including resources it contains
+      azure.azcollection.azure_rm_resourcegroup:
+        name: "{{ azure_virtual_machine_with_public_ip_resource_group }}"
+        force_delete_nonempty: true
+        state: absent
+
+- name: Test azure_virtual_machine_with_public_ip role with pre-created NIC and availability set in existent resource group
+  block:
+    - name: Set vars for resource creation
+      ansible.builtin.set_fact:
+        vnet_name: 'vnet-vm'
+        subnet_name: 'subnet-vm'
+        nic_name: 'NIC-vm'
+        availability_set_name: 'availability-set-vm'
+        public_ip_name: 'public-ip-vm'
+
+    - name: Create Networking stack
+      ansible.builtin.include_role:
+        name: cloud.azure_ops.azure_manage_networking_stack
+      vars:
+        azure_manage_networking_stack_operation: create
+        azure_manage_networking_stack_resource_group: "{{ resource_group }}"
+        azure_manage_networking_stack_virtual_network: "{{ vnet_name }}"
+        azure_manage_networking_stack_subnet: "{{ subnet_name }}"
+        azure_manage_networking_stack_vnet_address_prefixes_cidr:
+          - "10.1.0.0/16"
+        azure_manage_networking_stack_subnet_address_prefixes_cidr:
+          - "10.1.0.0/24"
+  
+    - name: Create Public ip
+      azure.azcollection.azure_rm_publicipaddress:
+        resource_group: "{{ resource_group }}"
+        name: "{{ public_ip_name }}"
+        allocation_method: static
+
+    - name: Create Network interface
+      ansible.builtin.include_role:
+        name: cloud.azure_ops.azure_manage_network_interface
+      vars:
+        azure_manage_network_interface_operation: create
+        azure_manage_network_interface_resource_group: "{{ resource_group }}"
+        azure_manage_network_interface_interface:
+          name: "{{ nic_name }}"
+          vnet_name: "{{ vnet_name }}"
+          subnet_name: "{{ subnet_name }}"
+          ip_configurations:
+            - name: "{{ public_ip_name }}"
+              primary: true
+
+    - name: Create availability set
+      azure.azcollection.azure_rm_availabilityset:
+        resource_group: "{{ resource_group }}"
+        name: "{{ availability_set_name }}"
+        platform_update_domain_count: 5
+        platform_fault_domain_count: 2
+
+    - name: Set vars for test
```

Comment:
From this point until "Delete Virtual machine" there are duplicated tasks
Can these tasks be in imported file? maybe can be included in "azure_vm_operations.yml"?

Reply:
Good point. Moved create, power_off, deallocate, power_on, restart tasks to a separate file